src/components/Layout/NavigationBar/NavigationBarItem.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from "react";
+import { FunctionComponent } from "react";
+import { NavLink } from "react-router-dom";
+import * as NavType from "./type";
+
+export const NavigationBarItem: FunctionComponent<{
+  item: NavType.NavigationItem;
+}> = ({ item }) => {
+  switch (item.schema) {
+    case NavType.NavigationItemType.IconButton:
+      return <IconButton item={item} />;
+    case NavType.NavigationItemType.LabelButton:
+      return <LabelButton item={item} />;
+    case NavType.NavigationItemType.DropDownList:
+      return <DropDownList item={item} />;
+    default:
+      return <NavigationLink item={item} />;
+  }
+};
+
+const NavigationLink: FunctionComponent<{ item: NavType.NavigationLink }> = ({ item }) => {
+  if (item.path.indexOf("http://") === 0 || item.path.indexOf("https://") === 0) {
+    return (
+      <a className={`font-medium ${item.className}`} href={item.path}>
+        {item.label}
+      </a>
+    );
+  } else {
+    return (
+      <NavLink className={`font-medium ${item.className}`} to={item.path}>
+        {item.label}
+      </NavLink>
+    );
+  }
+};
+
+const LabelButton: FunctionComponent<{ item: NavType.LabelButton }> = ({ item }) => {
+  if (item.path.indexOf("http://") === 0 || item.path.indexOf("https://") === 0) {
+    return (
+      <a href={item.path} className="w-full">
+        <button className={`font-bold py-2 px-3 ${item.className}`} data-testid={item.id}>
+          {item.label}
+        </button>
+      </a>
+    );
+  } else {
+    return (
+      <NavLink to={item.path} className="w-full">
+        <button className={`font-bold py-2 px-3 ${item.className}`} data-testid={item.id}>
+          {item.label}
+        </button>
+      </NavLink>
+    );
+  }
+};
+
+const IconButton: FunctionComponent<{ item: NavType.IconButton }> = ({ item }) => {
+  const ButtonIcon = item.icon;
+  if (item.path.indexOf("http://") === 0 || item.path.indexOf("https://") === 0) {
+    return (
+      <a className={`font-medium ${item.className}`} href={item.path} data-testid={item.id}>
+        <ButtonIcon />
+      </a>
+    );
+  } else {
+    return (
+      <NavLink className={`font-medium ${item.className}`} to={item.path} data-testid={item.id}>
+        <ButtonIcon />
+      </NavLink>
+    );
+  }
+};
+
+const DropDownList: FunctionComponent<{ item: NavType.DropDownList }> = ({ item }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <div className="relative">
```

Comment:
im thinking the dropdown list logic should on hover for `lg` upwards, the click (tap) for downwards 😬  otherwise this behaviour is fine for now i guess

Reply:
mm if hover will require 2 logic. One for desktop and other for mobile